src/include/duckdb/planner/column_binding.hpp

```diff
@@ -16,14 +16,18 @@ namespace duckdb {
 struct ColumnBinding {
 	idx_t table_index;
 	idx_t column_index;
+	idx_t lambda_index;
```

Comment:
Modifying the `ColumnBinding` seems rather invasive - as this is used in many different places in the code. Perhaps a cleaner solution would be to create a new expression type for lambda parameters - e.g. a `BoundLambdaRefExpression` - and use that instead of the `BoundColumnRefExpression`.

Reply:
Ah yes, that sounds much cleaner. 👍